### PR TITLE
[FEATURE] Allow running npm based quality checks via `runTests.sh`

### LINKED ISSUE #972: Allow running the JavaScript quality checks via `runTests.sh`

### PR DESCRIPTION
**Pull request by cumuru**
Additionally, a generic npm command has been added. 
See `runTests.sh --help` to explore the new commands `npm`, `lintStyle` and `lintJs`

Fixes #972

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,10 +14,10 @@
...
 	},
 	"scripts": {
-		"lint:js": "eslint 'Resources/Public/**/*.js'",
-		"lint:js:fix": "eslint 'Resources/Public/**/*.js' --quiet --fix",
-		"lint:style": "stylelint Resources/Public/**/*.css",
-		"lint:style:fix": "stylelint Resources/Public/**/*.css --fix"
+		"ci:lint:js": "eslint 'Resources/Public/**/*.js'",
+		"fix:lint:js": "eslint 'Resources/Public/**/*.js' --quiet --fix",
+		"ci:lint:style": "stylelint Resources/Public/**/*.css",
+		"fix:lint:style": "stylelint Resources/Public/**/*.css --fix"
```
Comment (oliverklee) [inline package.json:20]: Please let's move the renaming to a separate pre-patch PR to follow the "only one thing per PR" principle. :-)

File: Build/Scripts/runTests.sh [hunk 2]
```diff
@@ -155,7 +155,9 @@ Options:
             - functional: PHP functional tests
             - lintTypoScript: TypoScript linting
             - lintPhp: PHP linting
+            - lintJs: JavaScript file linting. Set -n for dry-run.
             - lintJson: JSON linting
+            - lintStyle: CSS file linting. Set -n for dry-run.
```
Comment (oliverklee) [inline Build/Scripts/runTests.sh:160]: I'd expect this command to be named `lintCss` instead to be consistent with the other commands that have the language/file type after "lint".

File: Build/Scripts/runTests.sh [hunk 3]
```diff
@@ -308,10 +310,14 @@ PHP_XDEBUG_ON=0
 PHP_XDEBUG_PORT=9003
 EXTRA_TEST_OPTIONS=""
 PHPUNIT_RANDOM=""
+# CGLCHECK_DRY_RUN is a more generic dry-run switch not limited to CGL
 CGLCHECK_DRY_RUN=0
 DATABASE_DRIVER=""
 CONTAINER_BIN=""
 COMPOSER_ROOT_VERSION="3.0.x-dev"
+NODE_VERSION=18
```
Comment (oliverklee) [inline Build/Scripts/runTests.sh:318]: Could we also use a newer version, e.g. 20 or 22?

File: Build/Scripts/runTests.sh [hunk 4]
```diff
@@ -631,8 +661,12 @@ if [[ ${IS_CORE_CI} -eq 1 ]]; then
 else
     echo "Environment: local" >&2
 fi
-echo "PHP: ${PHP_VERSION}" >&2
-echo "TYPO3: ${CORE_VERSION}" >&2
+if [[ ${TEST_SUITE} =~ ^npm$ ]]; then
+    echo "NODE: ${NODE_VERSION}" >&2
```
Comment (oliverklee) [inline Build/Scripts/runTests.sh:665]: For both commands, I still see the PHP and TYPO3 versions:

```
Now running 'npm ci --silent'.

> tea@1.0.0 fix:lint:js
> eslint 'Resources/Public/**/*.js' --quiet --fix


###########################################################################
Result of lintJs
Environment: local
PHP: 8.3
TYPO3: 12.4
CONTAINER_BIN: podman
SUCCESS
###########################################################################
```

This doesn't look intentional to me.

**Review by oliverklee — CHANGES_REQUESTED** [review]

File: Build/Scripts/runTests.sh (hunk 3 above)
Comment (cumuru) [reply Build/Scripts/runTests.sh:318]: The package.json doesn‘t allow any higher version, so that‘s a different story.

Comment (oliverklee) [reply Build/Scripts/runTests.sh:319]: Ack.

**Comment by oliverklee** [comment]
I've just merged #1404. So this PR now needs a rebase.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -88,7 +88,7 @@ jobs:
       fail-fast: false
       matrix:
         command:
-          - "style"
+          - "css"
```
Comment (oliverklee) [inline .github/workflows/ci.yml:91]: I have just moved this renaming to a separate PR: #1406
